src/internal/signalr/handler.go

```diff
@@ -62,9 +74,53 @@ func NewObservationsHandler(thing adapter.Thing, cache cache.Cache) (Handler, er
 		model.LockCablePermanently:  handler.handleLockCablePermanently,
 	}
 
+	go handler.manageEnergyObservation()
+
 	return &handler, nil
 }
 
+func (h *observationsHandler) manageEnergyObservation() {
+	meterElecSrv, err := h.getMeterElecService()
+	if err != nil {
+		log.Error(err)
+	}
+
+	var prevValue model.Observation
+
+	for {
+		select {
+		case val := <-h.energyObservationChan:
+			h.lock.Lock()
+
+			if val.Timestamp.Before(prevValue.Timestamp) {
+				return
+			}
+
+			v, err := val.Float64Value()
+			if err != nil {
+				log.WithError(err)
+			}
+
+			h.cache.SetLifetimeEnergy(v)
```

Comment:
You should store the value in an intermediate variable. If someone would send a command to get meter readings, we might have inconsistent data (again).